src/lib/symbol_overrides.cc

```diff
@@ -93,33 +252,259 @@ struct free {
   static inline bool ref_checked = false;
 
   static void hook(void *ptr) noexcept {
-    check_libraries();
+    DeallocTrackerHelper helper;
     if (ptr == nullptr) {
       return;
     }
 
-    ddprof::AllocationTracker::track_deallocation_s(
-        reinterpret_cast<uintptr_t>(ptr));
+    helper.track(ptr);
     ref(ptr);
   }
 };
 
+struct free_sized {
+  static constexpr auto name = "free_sized";
+  static inline auto ref = &::free_sized;
+  static inline bool ref_checked = false;
+
+  static void hook(void *ptr, size_t size) noexcept {
```

Comment:
Is the size given always the same as the size from the allocation ?

Reply:
This is not relevant to the PR, I'm mostly worried about live heap's accuracy.

Reply:
it's the responsibility of the caller to give the same size as the one passed to the allocation function.